link/link.go

```diff
@@ -355,32 +353,68 @@ func (l *RawLink) Info() (*Info, error) {
 	var extra interface{}
```

Comment:
hum, so the idea is to run the link info syscall twice, first to get the link info type and second to read it into specific type?

could we just store the link type, like to have Link interface function that returns that? seems wasteful to do 2 same syscalls just to get it into convenient type

Reply:
@rgo3 @lmb I updated the missed counts interface changes https://github.com/cilium/ebpf/pull/1295 on top of this change, still need to update the tetragon code on top of that, but so far looks good

Reply:
We can move the `info.Type` specific code from this function to new `Info()` methods on the concrete link types, like you did with `KprobeMulti.Info()` before. Then the only caller of `RawLink.Info()` is `wrapRawLink` which only really cares about `info.Type` anyways. This way `Info()` only does a single call in the common case (caveat: sometimes we need two calls anyways to retrieve buffer sizes, etc.).

We can make that change in this PR, or as a follow up, or once you've landed your kprobe missed count changes. Up to you and @rgo3.

Reply:
How would having separate `Info()` methods on all concrete link types as well look in practice when loading a pinned link from bpffs and get specific information from it?

1. Load pinned link
2. Call generic Info method
3. Check type of returned object that satisfies the `Link` interface
4. Assert concrete link type
5. Call type dependent Info method

Reply:
That sounds right to me. 1-4 happen behind the scenes in LoadPinnedLink. 5 happens when a user does `KprobeMulti.Info()`.